listener.go

```diff
@@ -25,8 +25,15 @@ const (
 )
 
 func processSnapshot(snapshot *common.Snapshot) {
-	log.Debugf("Snapshot received. Switching to DB version: %s", snapshot.SnapshotInfo)
 
+	if apidInfo.LastSnapshot != "" && apidInfo.LastSnapshot != snapshot.SnapshotInfo {
+		log.Debugf("Release snapshot for {%s}. Switching to version {%s}",
+			apidInfo.LastSnapshot , snapshot.SnapshotInfo)
+		dataService.ReleaseDB(apidInfo.LastSnapshot)
+	} else {
+		log.Debugf("Process snapshot for version {%s}",
+			snapshot.SnapshotInfo)
+	}
```

Comment:
It would be better to move this code block behind the `processSqliteSnapshot(db)` for the following reasons:
1. We want to release&delete a db only after the new snapshot has been successfully verified&processed. Otherwise, if APID crushes during processing a new snapshot, we may not be able to debug it because the old DBs have been deleted.
2. Other plugins like apidVerifyApiKey or apidGatewayConfDeploy may be reading/writing in parallel while ApigeeSync is switching to a new snapshot. If we delete old one first, there may be a short interval before switching to a new one, which may cause operation DB operation failures in other plugins. That's a race.

Reply:
Actually it will have to after setDB(db).

Reply:
Yes, good point.

Reply:
done.